### PR TITLE
Task/DR 690 axe report test

### PR DESCRIPTION
**Pull request by timaNHS**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/frontend/.prettierignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (timrickwood) [inline src/frontend/.prettierignore:1]: Can't comment against the empty gitleaks.tom file ? What is this ? We have a .gitleaks.toml file ? Is this something else ?

File: tests/ui/src/pages/BasePage.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// import { Page } from 'playwright';
```
Comment (timrickwood) [inline tests/ui/src/pages/BasePage.ts:1]: If this is all commented out is it needed ?

File: src/frontend/.prettierignore (hunk 1 above)
Comment (timaNHS) [reply src/frontend/.prettierignore:1]: Yes, this is a utility to format the files.

**Comment by tamaragoldschmidt** [comment]
The branch and the PR description don't match. Is this the correct branch?

**Comment by timaNHS** [comment]
> The branch and the PR description don't match. Is this the correct branch?

yes, I based the changes on your branch, keeping the different name

File: .gitleaks.toml [hunk 3]
```diff
@@ -7,6 +7,10 @@ useDefault = true # SEE: https://github.com/gitleaks/gitleaks/blob/master/config
 description = "allow ="
 Regexes = '''========================================='''
 
+[[allowlist.files]]
+pattern = '''^tests/ui/artifacts/accessibilityReport\.html$'''
```
Comment (tamaragoldschmidt) [inline .gitleaks.toml:11]: This file should be in gitignore not in gitleaks

File: src/frontend/.prettierrc [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (tamaragoldschmidt) [inline src/frontend/.prettierrc:1]: Again - does this file relate tot he UI or the tests?

File: tests/ui/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (tamaragoldschmidt) [inline tests/ui/tsconfig.json:1]: what is this file required for?

File: tests/ui/test/features/search.feature [hunk 6]
```diff
@@ -1,5 +1,13 @@
 Feature: Cloudfront tests
 
+@Test
```
Comment (tamaragoldschmidt) [inline tests/ui/test/features/search.feature:3]: Can you remove this tag

File: tests/ui/test/features/search.feature [hunk 7]
```diff
@@ -1,5 +1,13 @@
...
   Scenario: Navigate to home page via env
     Given I navigate to the cloudfront endpoint
     Then "Capacity Management" link is displayed on the page
+    And the accessibility checks are passing
+
+  @Test
```
Comment (tamaragoldschmidt) [inline tests/ui/test/features/search.feature:9]: can you remove this tag

File: tests/ui/src/utilities/accessibility.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  async runAxeCheck(testId: string) {
+    const accessibilityScanResults = await new AxeBuilder({
+      page: pageFixture.page,
+    }).analyze();
+
+    createHtmlReport({
+      results: accessibilityScanResults,
+      options: {
+        projectKey: "uec-cm-" + testId,
+        outputDirPath: "artifacts/" + testId,
```
Comment (tamaragoldschmidt) [inline tests/ui/src/utilities/accessibility.ts:20]: Have you checked the path that this creates?

File: tests/ui/src/utilities/accessibility.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+      },
+    });
+
+    return accessibilityScanResults;
+  }
+}
```
Comment (tamaragoldschmidt) [inline tests/ui/src/utilities/accessibility.ts:26]: I thought that there was going to be a error on x numbers of violations check in here

File: tests/ui/test/features/search.feature [hunk 10]
```diff
@@ -1,5 +1,13 @@
...
+  Scenario: Removing accessible element fails the test
+    Given I navigate to the cloudfront endpoint
+    When I remove any input element
+    Then the accessibility checks are failing
```
Comment (tamaragoldschmidt) [inline tests/ui/test/features/search.feature:13]: Does this test pass for you?

File: tests/ui/src/utilities/accessibility.ts (hunk 8 above)
Comment (timaNHS) [reply tests/ui/src/utilities/accessibility.ts:20]: yes :)

File: tests/ui/src/utilities/accessibility.ts (hunk 9 above)
Comment (timaNHS) [reply tests/ui/src/utilities/accessibility.ts:26]: it just returns a FAIL on the single test

File: tests/ui/test/features/search.feature (hunk 10 above)
Comment (timaNHS) [reply tests/ui/test/features/search.feature:13]: no, it's meant to fail which it does

File: tests/ui/tsconfig.json (hunk 5 above)
Comment (timaNHS) [reply tests/ui/tsconfig.json:1]: the reference to document fails on ` const element = document.querySelector(selector);`, which can only be fixed by specifying a tsconfig localised to the test/ui folder

File: src/frontend/.prettierrc (hunk 4 above)
Comment (timaNHS) [reply src/frontend/.prettierrc:1]: yes, there were a few formatting inconsistencies with the quotes, tabs, and alignments which prettier fixes

File: tests/ui/test/features/search.feature (hunk 7 above)
Comment (timaNHS) [reply tests/ui/test/features/search.feature]: removed on [423c36c](https://github.com/NHSDigital/uec-cm/pull/80/commits/423c36c92e4e3b169d1bf4f7fa2b531700d9add6)

File: tests/ui/test/features/search.feature (hunk 6 above)
Comment (timaNHS) [reply tests/ui/test/features/search.feature]: removed on [423c36c](https://github.com/NHSDigital/uec-cm/pull/80/commits/423c36c92e4e3b169d1bf4f7fa2b531700d9add6)

File: .gitleaks.toml (hunk 3 above)
Comment (timaNHS) [reply .gitleaks.toml]: removed on [423c36c](https://github.com/NHSDigital/uec-cm/pull/80/commits/423c36c92e4e3b169d1bf4f7fa2b531700d9add6)

File: src/frontend/.prettierignore (hunk 1 above)
Comment (timaNHS) [reply src/frontend/.prettierignore:1]: Both really, the `package.json` is in the ui folder

File: tests/ui/test/features/search.feature (hunk 10 above)
Comment (tamaragoldschmidt) [reply tests/ui/test/features/search.feature:11]: I would have expected the test to pass as it is checking for a faijlure

**Comment by gdcNHS** [comment]
Work has been superseded
